lib/Config/FetcherConfig.php

```diff
@@ -94,6 +94,28 @@ public function __construct(IConfig $config)
         return $this;
     }
 
+    /**
+     * Checks for available encoding options
+     *
+     * @return String list of supported encoding types
+     */
+    public function checkEncoding()
+    {
+        $supportedEncoding = [];
+
+        // check curl features
+        $curl_features = curl_version()["features"];
+
+        $bitfields = array('CURL_VERSION_LIBZ' => ['gzip', 'deflate'], 'CURL_VERSION_BROTLI' => ['br']);
+        
+        foreach ($bitfields as $feature => $header) {
+            // checking available features via the 'features' bitmask and adding available types to the list
+            $curl_features & constant($feature) ?
+                $supportedEncoding = array_merge($supportedEncoding, $header) : null;
```

Comment:
Checks if the feature exists and is a little easier to read.
```suggestion
            if (defined($feature) && $curl_features & constant($feature)) {
              $supportedEncoding = array_merge($supportedEncoding, $header);
            }
```